src/theory/arrays/theory_arrays.cpp

```diff
@@ -1252,9 +1252,9 @@ bool TheoryArrays::collectModelInfo(TheoryModel* m)
 void TheoryArrays::presolve()
 {
   Trace("arrays")<<"Presolving \n";
-  // add the decision strategy
-  getDecisionManager()->registerStrategy(DecisionManager::STRAT_ARRAYS,
-                                         d_dstrat.get());
+  // add the decision strategy, which is user-context-independent
+  getDecisionManager()->registerStrategy(
+      DecisionManager::STRAT_ARRAYS, d_dstrat.get(), true);
```

Comment:
shouldn't the third argument here be `false`, which indicates context-independent?

Reply:
Yes, you're right, good catch.

Notice that in all three of these cases, the code is robust to the user-context-dependent-ness of the strategy, since it is always re-registered during presolve calls.

Reply:
Actually it was a bit more complicated than i thought, we need a third option (strategies that are local to a solve scope).  I've made an enum in my latest commit, PTAL.